tools/android/packaging/xbmc/build.gradle.in

```diff
@@ -1,26 +1,35 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 29
-    buildToolsVersion "29.0.3"
+    compileSdkVersion 31
     defaultConfig {
         applicationId "@APP_PACKAGE@"
         minSdkVersion 21
         targetSdkVersion 29
         versionCode @APP_VERSION_CODE_ANDROID@
         versionName "@APP_VERSION@"
     }
-    buildTypes {
-        releaseUnsigned.initWith(buildTypes.release)
-        releaseUnsigned {
-            signingConfig null
+    signingConfigs {
+        release {
+            keyAlias System.getenv("KODI_ANDROID_KEY_ALIAS")
+            keyPassword System.getenv("KODI_ANDROID_KEY_PASSWORD")
+            storeFile file(System.getenv("KODI_ANDROID_STORE_FILE"))
+            storePassword System.getenv("KODI_ANDROID_STORE_PASSWORD")
+            enableV1Signing true
+            enableV2Signing true
+            enableV3Signing true
```

Comment:
Can you explain what this is and why we want it?

Reply:
Android has incremental signing schemes that bring in new features/securities. jarsigner is only generic and not android specific, so it only does v1 signing. The transition to apksigner (either explicitly or gradle uses internally) brings the opportunity to sign using the newer schemes with no change to the end user.

V2 scheme brings in the more "secure" signing. v3's major feature is about key rotation. Only good things really, even if we dont use them (ie key rotation).

v4 scheme starts to bring in more changes like differentiation between upload signing and release signing. However it produces an idsig file. Until we need to go with v4, i dont think dealing with that is worth the hassle right now.

Reply:
I should mention also, if a device runtime doesnt support a newer signing mechanism, it just falls back to the lower level.

Reply:
just reread this, and possibly you are asking about the syntax change. Somewhere i think in gradle 4 the syntax for signing changed from the one you had implemented into this. v3 scheme was done after the transition, so there is no v3SigningEnabled, hence the transition to the new style to support all currently available schems